devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -51,423 +46,137 @@ metadata {
-        value: value,
-        descriptionText: "${linkText} level is ${value}%"
-    ]
-}
-
-private Map makePressureResult(rawValue) {
-    log.debug 'makePressureResut'
-    def linkText = getLinkText(device)
-
-    def pascals = rawValue / 10
-    def result = [
-        name: 'pressure',
-        descriptionText: "${linkText} pressure is ${pascals}Pa",
-        value: pascals
-    ]
-
     return result
 }
 
-private Map makeBatteryResult(rawValue) {
-    // log.debug 'makeBatteryResult'
-    def linkText = getLinkText(device)
-
-    // log.debug
-    [
-        name: 'battery',
-        value: rawValue,
-        descriptionText: "${linkText} battery is at ${rawValue}%"
-    ]
-}
-
-private Map makeTemperatureResult(value) {
-    // log.debug 'makeTemperatureResult'
-    def linkText = getLinkText(device)
-
-    // log.debug "tempOffset: ${tempOffset}"
-    if (tempOffset) {
-        def offset = tempOffset as int
-        // log.debug "offset: ${offset}"
-        def v = value as int
-        // log.debug "v: ${v}"
-        value = v + offset
-        // log.debug "value: ${value}"
-    }
-
-    return [
-        name: 'temperature',
-        value: "" + value,
-        descriptionText: "${linkText} is ${value}°${temperatureScale}",
-        unit: temperatureScale
-    ]
-}
-
-/**** HELPER METHODS ****/
-private def convertTemperatureHex(value) {
-    // log.debug "convertTemperatureHex(${value})"
-    def celsius = Integer.parseInt(value, 16).shortValue() / 100
-    // log.debug "celsius: ${celsius}"
-
-    return convertTemperature(celsius, "C")
-}
-
-private def convertTemperature(value, scale = "C") {
-    if(getTemperatureScale() == scale){
-        return Math.round(value * 100) / 100
-    } else {
-        if (scale == "C") {
-            // Celsius to Fahrenheit
-            return Math.round(celsiusToFahrenheit(value) * 100) /100
-        }
-        // Fahrenheit to Celsius
-        return Math.round(fahrenheitToCelsius(value) * 100) /100
-    }
-}
-
-private def makeSerialResult(serial) {
-    log.debug "makeSerialResult: " + serial
-
-    def linkText = getLinkText(device)
-    sendEvent([
-        name: "serial",
-        value: serial,
-        descriptionText: "${linkText} has serial ${serial}" ])
-    return [
-        name: "serial",
-        value: serial,
-        descriptionText: "${linkText} has serial ${serial}" ]
-}
-
-// takes a level from 0 to 100 and translates it to a ZigBee move to level with on/off command
-private def makeLevelCommand(level) {
-    def rangeMax = 254
-    def scaledLevel = Math.round(level * rangeMax / 100)
-    log.debug "scaled level for ${level}%: ${scaledLevel}"
-
-    // convert to hex string and pad to two digits
-    def hexLevel = new BigInteger(scaledLevel.toString()).toString(16).padLeft(2, '0')
-
-    "st cmd 0x${device.deviceNetworkId} 1 8 4 {${hexLevel} 0000}"
+def getPressureResult(rawValue) {
+    def kpa = rawvalue / (10 * 1000) // reports are in deciPascals, I think
```

Comment:
deciPascals is correct I believe. I compared the value reported by this attribute to the value reported by the standard MeasuredValue attribute:

0x0020: 979339
0x0000: 979

The definition of attribute 0x0000 is the pressure in kPa times 10